metadata-ingestion/src/datahub/ingestion/source/usage/starburst_trino_usage.py

```diff
@@ -0,0 +1,248 @@
+                    yield wu
+
+        def _make_usage_query(self) -> str:
+            return trino_usage_sql_comment.format(
+                audit_catalog=self.config.audit_catalog,
+                audit_schema=self.config.audit_schema,
+                start_time=self.config.start_time.strftime(trino_datetime_format),
+                end_time=self.config.end_time.strftime(trino_datetime_format),
+            )
+
+        def _make_sql_engine(self) -> Engine:
+            url = self.config.get_sql_alchemy_url()
+            logger.debug(f"sql_alchemy_url = {url}")
+            engine = create_engine(url, **self.config.options)
+            return engine
+
+        def _get_trino_history(self):
+            query = self._make_usage_query()
+            engine = self._make_sql_engine()
+            results = engine.execute(query)
+            events = []
+            for row in results:
+                # minor type conversion
+                if hasattr(row, "_asdict"):
+                    event_dict = row._asdict()
+                else:
+                    event_dict = dict(row)
+
+                # stripping extra spaces caused by above _asdict() conversion
+                for k, v in event_dict.items():
+                    if isinstance(v, str):
+                        event_dict[k] = v.strip()
+
+                if event_dict.get("starttime", None):
+                    event_dict["starttime"] = event_dict.get("starttime").__str__()
+                if event_dict.get("endtime", None):
+                    event_dict["endtime"] = event_dict.get("endtime").__str__()
+
+                logger.debug(f"event_dict: {event_dict}")
+                events.append(event_dict)
+
+            if events:
+                return events
+
+            # SQL results can be empty. If results is empty, the SQL connection closes.
+            # Then, we don't want to proceed ingestion.
+            logging.info("SQL Result is empty")
+            return None
+
+        def _convert_str_to_datetime(self, v):
+            if isinstance(v, str):
+                isodate = parser.parse(v)  # compatible with Python 3.6+
+                return isodate
+
+        def _get_joined_access_event(self, events):
+            joined_access_events = []
+            for event_dict in events:
+                event_dict["create_time"] = self._convert_str_to_datetime(
+                    event_dict.get("create_time")
+                )
+
+                event_dict["end_time"] = self._convert_str_to_datetime(
+                    event_dict.get("end_time")
+                )
+
+                if not event_dict["accessed_metadata"]:
+                    logging.info("Field accessed_metadata is empty. Skipping ....")
+                    continue
+
+                event_dict["accessed_metadata"] = json.loads(
+                    event_dict["accessed_metadata"]
+                )
+
+                if not event_dict.get("usr") or event_dict["usr"] == "":
```

Comment:
can simplify with `if not event_dict.get("usr"):`  
since `if not ""` evaluates to True.